### PR TITLE
*: fix upgrade error for version 181

### LINKED ISSUE #49485: tidb launch failed when upgarding
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
[2023/12/15 11:28:52.532 +08:00] [FATAL] [terror.go:309] ["unexpected error"] [error="[planner:1054]Unknown column 'type' in 'field list'"] [stack="github.com/pingcap/tidb/pkg/parser/terror.MustNil\n\t/Users/wangchao/Code/pingcap/tidb/pkg/parser/terror/terror.go:309\nmain.createStoreAndDomain\n\t/Users/wangchao/Code/pingcap/tidb/cmd/tidb-server/main.go:403\nmain.main\n\t/Users/wangchao/Code/pingcap/tidb/cmd/tidb-server/main.go:302\nruntime.main\n\t/Users/wangchao/.gvm/gos/go1.21.0/src/runtime/proc.go:267"] [stack="github.com/pingcap/tidb/pkg/parser/terror.MustNil\n\t/Users/wangchao/Code/pingcap/tidb/pkg/parser/terror/terror.go:309\nmain.createStoreAndDomain\n\t/Users/wangchao/Code/pingcap/tidb/cmd/tidb-server/main.go:403\nmain.main\n\t/Users/wangchao/Code/pingcap/tidb/cmd/tidb-server/main.go:302\nruntime.main\n\t/Users/wangchao/.gvm/gos/go1.21.0/src/runtime/proc.go:267"]
```

### 2. What did you expect to see? (Required)

no error

### 3. What did you see instead (Required)

error occurs when launch tidb

### 4. What is your TiDB version? (Required)

master

### PR DESCRIPTION
**Pull request by lcwangchao**
### What problem does this PR solve?

Issue Number: close #49485

Problem Summary:

We did not assig a explicitly version to `version181`, see: https://github.com/pingcap/tidb/pull/49426/files#diff-2fde3341e0231e7b162880b36f07d23b7949ca512c5536b50706ac5aa9072b25R1045

Seems golang will not increase the last const value when the list not started with iota, see: https://go.dev/play/p/dCORpY3lJP7

```
const (
	v1 = 1
	v2
)

func main() {
	fmt.Println(v1)
	fmt.Println(v2)
}
```

result is

```
1
1
```

### What changed and how does it work?

See PR

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by YangKeao — APPROVED** [review]

**Review by tiancaiamao — APPROVED** [review]

**Comment by lance6716** [comment]
Do you have time to update the linter to detect this issue?

build/linter/bootstrap/analyzer.go

File: build/linter/bootstrap/analyzer.go [hunk 1]
```diff
@@ -89,20 +90,44 @@ func run(pass *analysis.Pass) (interface{}, error) {
...
+							continue
+						}
+
+						valInName, err := strconv.Atoi(name[len("version"):])
+						if err != nil {
+							continue
+						}
+
+						if len(v2.Values) != 1 {
+							panic(fmt.Sprintf("bootstrap.go: the value of version variable '%s' must be specified explicitly", name))
```
Comment (lance6716) [inline build/linter/bootstrap/analyzer.go:112]: we can use `pass.Reportf` like below lines. It can report the line/column number of golang AST

**Review by lance6716 — COMMENTED** [review]
rest lgtm

**Review by lance6716 — APPROVED** [review]

**Pull request merged**